client/lib/interval/runner.js

```diff
@@ -107,43 +107,65 @@ export function remove( id ) {
 }
 
 function removeFromQueue( id ) {
-	state = state.update( 'actions', removeFromList( id ) );
+	state = {
+		...state,
+		actions: removeFromList( id )( state.actions ),
+	};
 }
 
 function getPeriodActions( period ) {
-	return state
-		.get( 'actions' )
-		.filter( a => a.get( 'period' ) === period );
+	return filter( state.actions, { period } );
 }
 
 function hasPeriodActions( period ) {
-	return state
-		.get( 'actions' )
-		.some( a => a.get( 'period' ) === period );
+	return some( state.actions, { period } );
 }
 
 function executePeriodActions( period ) {
 	// Make sure we don't return `false` or it will
 	// halt the iteration in `forEach`
-	const callAction = a => a.get( 'onTick' ).call() || true;
+	const callAction = a => a.onTick.call() || true;
 
 	getPeriodActions( period ).forEach( callAction );
 
-	state = state.setIn( [ 'periodTimers', period ], null );
+	state = {
+		...state,
+		periodTimers: {
+			...state.periodTimers,
+			[ period ]: null,
+		},
+	};
```

Comment:
would `lodash.merge()` make this clearer here?

Reply:
maybe? it would be:

```
state = merge( 
  {}, 
  state, 
  { 
    periodTimers: {
      [ p ]: null
    }
  }
); 
```

i think?

Reply:
huh…well when you put it that way…maybe not